lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriterOptions.java

```diff
@@ -13,52 +13,98 @@
 import static java.lang.Math.toIntExact;
 import static java.util.Objects.requireNonNull;
 
 public class ParquetWriterOptions
 {
     private static final DataSize DEFAULT_MAX_ROW_GROUP_SIZE = DataSize.ofBytes(ParquetWriter.DEFAULT_BLOCK_SIZE);
     private static final DataSize DEFAULT_MAX_PAGE_SIZE = DataSize.ofBytes(ParquetWriter.DEFAULT_PAGE_SIZE);
+    private static final double DEFAULT_BLOOM_FILTER_FPP = 0.05;
 
     public static ParquetWriterOptions.Builder builder()
     {
         return new ParquetWriterOptions.Builder();
     }
 
-    private final int maxRowGroupSize;
-    private final int maxPageSize;
+    private final DataSize maxRowGroupSize;
+    private final DataSize maxPageSize;
+    private final double bloomFilterFpp;
+    private final Set<String> bloomFilterColumns;
 
-    private ParquetWriterOptions(DataSize maxBlockSize, DataSize maxPageSize)
+    private ParquetWriterOptions(DataSize maxBlockSize, DataSize maxPageSize, Set<String> bloomFilterColumns, double bloomFilterFpp)
     {
-        this.maxRowGroupSize = toIntExact(requireNonNull(maxBlockSize, "maxBlockSize is null").toBytes());
-        this.maxPageSize = toIntExact(requireNonNull(maxPageSize, "maxPageSize is null").toBytes());
+        this.maxRowGroupSize = requireNonNull(maxBlockSize, "maxBlockSize is null");
+        this.maxPageSize = requireNonNull(maxPageSize, "maxPageSize is null");
+        this.bloomFilterColumns = ImmutableSet.copyOf(bloomFilterColumns);
+        this.bloomFilterFpp = bloomFilterFpp;
     }
 
     public long getMaxRowGroupSize()
     {
-        return maxRowGroupSize;
+        return toIntExact(maxRowGroupSize.toBytes());
     }
 
     public int getMaxPageSize()
     {
-        return maxPageSize;
+        return toIntExact(maxPageSize.toBytes());
+    }
+
+    public boolean isBloomFilterEnabled()
+    {
+        return !bloomFilterColumns.isEmpty();
+    }
+
+    public boolean isBloomFilterColumn(String columnName)
+    {
+        return bloomFilterColumns.contains(columnName);
+    }
+
+    public double getBloomFilterFpp()
+    {
+        return this.bloomFilterFpp;
+    }
+
+    public Set<String> getBloomFilterColumns()
+    {
+        return this.bloomFilterColumns;
     }
 
     public static class Builder
     {
         private DataSize maxBlockSize = DEFAULT_MAX_ROW_GROUP_SIZE;
         private DataSize maxPageSize = DEFAULT_MAX_PAGE_SIZE;
+        private double bloomFilterFpp = DEFAULT_BLOOM_FILTER_FPP;
+        private Set<String> bloomFilterColumns = ImmutableSet.of();
 
         public Builder setMaxBlockSize(DataSize maxBlockSize)
         {
             this.maxBlockSize = maxBlockSize;
             return this;
         }
 
+        public Builder setBloomFilterColumns(Set<String> bloomFilterColumns)
+        {
+            this.bloomFilterColumns = bloomFilterColumns;
+            return this;
+        }
+
+        public Builder setBloomFilterFpp(double bloomFilterFpp)
```

Comment:
Rename variable to `fpp` or simply `value` for conciseness. It's understood that this is the value of the bloom filter fpp

Reply:
Actually I agree that those are better names, but the OrcWriterOptions already uses `bloomFilterFpp` name. https://github.com/trinodb/trino/blob/fe608f2723842037ff620d612a706900e79c52c8/lib/trino-orc/src/main/java/io/trino/orc/OrcWriterOptions.java#L232